### PR TITLE
feat: Update ios product price on price change from discovery

### PR DESCRIPTION
**Pull request by jawad-khan**
# ⛔️ MAIN BRANCH WARNING! 2U EMPLOYEES must make branches against the 2u/main BRANCH

- [ ] I have checked the branch to which I would like to merge.

# ⛔️ DEPRECATION WARNING

**This repository is deprecated and in maintainence-only operation while we work on a replacement, please see [this announcement](https://discuss.openedx.org/t/deprecation-removal-ecommerce-service-depr-22/6839) for more information.**

Although we have stopped integrating new contributions, we always appreciate security disclosures and patches sent to [security@edx.org](mailto:security@edx.org)

## Required Testing
- [ ] Before deploying this change, complete a purchase in the stage environment. 
(^ We can remove that manual check once REV-2624 is done and the corresponding e2e test runs again)

## Description
This PR updates mobile seats in ecommerce and on ios store whenever there is a price or expiry change pushed through discovery. Since we are handling android products manually right now, it sends email notification to mobile team about this change. This PR also set ios product availability for all regions.
https://2u-internal.atlassian.net/browse/LEARNER-9776

Useful information to include:
- This change will effect discovery job which updates ecommerce and mobile users.
- Include screenshots for changes to the UI (ideally, both "before" and "after" screenshots, if applicable).
- Provide links to the description of corresponding configuration changes. Remember to correctly annotate these changes.

## Supporting information
https://2u-internal.atlassian.net/browse/LEARNER-9776

## Testing instructions

Please provide detailed step-by-step instructions for testing this change; how did YOU test this change?

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, OpenEdx vs. edx.org differences, development vs. production environment differences, security, or accessibility.

### CODE REVIEW

**Review by moeez96 — APPROVED** [review]
Overall LGTM 👍

File: ecommerce/extensions/iap/api/v1/utils.py [hunk 1]
```diff
@@ -286,6 +288,42 @@ def upload_screenshot_of_inapp_purchase(in_app_purchase_id, headers):
         raise AppStoreRequestException("Couldn't finalize screenshot")
 
 
+def set_territories_of_in_app_purchase(in_app_purchase_id, headers):
+    url = 'https://api.appstoreconnect.apple.com/v1/territories?limit=200'
```
Comment (julianajlk) [inline ecommerce/extensions/iap/api/v1/utils.py]: Can you use `APP_STORE_BASE_URL` for this URL following the same pattern of the other URLs in this file

File: ecommerce/extensions/iap/api/v1/utils.py [hunk 2]
```diff
@@ -286,6 +288,42 @@ def upload_screenshot_of_inapp_purchase(in_app_purchase_id, headers):
...
+    response = request_connect_store(url, headers, method='get')
+    if response.status_code != 200:
+        raise AppStoreRequestException("Couldn't fetch territories")
+
+    territories = [{'type': territory['type'], 'id': territory['id']}
+                   for territory in response.json()['data']]
+
+    url = 'https://api.appstoreconnect.apple.com/v1/inAppPurchaseAvailabilities'
```
Comment (julianajlk) [inline ecommerce/extensions/iap/api/v1/utils.py]: Same for this URL

File: ecommerce/extensions/api/utils.py [hunk 3]
```diff
@@ -32,5 +32,8 @@ def format_seat(seat):
         'body': "\n".join(formatted_seats)
     }
 
+    if failure_msg:
+        messages['body'] += "\n Failed to update above mobile seats, please do it manually."
```
Comment (julianajlk) [inline ecommerce/extensions/api/utils.py:36]: Is `messages['body']` a string?

**Review by julianajlk — APPROVED** [review]

File: ecommerce/extensions/api/utils.py (hunk 3 above)
Comment (jawad-khan) [reply ecommerce/extensions/api/utils.py:36]: yes, it is.

**Pull request merged**